### PR TITLE
reference_builder: use `normalize_author_name`.

### PR DESCRIPTION
**Pull request by ammirate**
Signed-off-by: Antonio Cesarano <cesarano2607@gmail.com>

Closes inspirehep/inspire-dojson#69

### CODE REVIEW

**Comment by michamos** [comment]
actually only `add_author` seems better, so that names get normalized no matter the origin.

**Comment by ammirate** [comment]
You mean not using `normalize_author_name` then extracting names in `_split_refextract_authors_str`, but only in `add_author`?

**Comment by ammirate** [comment]
@michamos do the tests look meaningful for you?

File: tests/unit/test_reference_builder.py [hunk 1]
```diff
@@ -452,6 +408,82 @@ def test_add_refextract_author_str():
...
+
+    builder.add_refextract_authors_str(
+        'S. B. Cenko, M. M. Kasliwal, D. A. Perley et al.'
+    )
+
+    expected = [
+        {
+            'reference': {
+                'authors': [
+                    {'full_name': 'Cenko, S. B.'},
```
Comment (michamos) [inline tests/unit/test_reference_builder.py]: this is strange, it should remove the spaces:
```
In [1]: from inspire_schemas.utils import normalize_author_name

In [2]: normalize_author_name('Cenko, S. B.')
Out[2]: u'Cenko, S.B.'
```

File: tests/unit/test_reference_builder.py [hunk 2]
```diff
@@ -452,6 +408,82 @@ def test_add_refextract_author_str():
...
+            },
+        },
+    ]
+    result = [builder.obj]
+
+    assert validate(result, subschema) is None
+    assert expected == result
+
+
+def test_add_refextract_authors_str_discards_et_al():
```
Comment (michamos) [inline tests/unit/test_reference_builder.py]: repeated function name.

File: tests/unit/test_reference_builder.py (hunk 1 above)
Comment (ammirate) [reply tests/unit/test_reference_builder.py]: True. Then, why the build passed? :/

Comment (michamos) [reply tests/unit/test_reference_builder.py]: becaues of the other issue, you repeated the function name, so you redefined the function, so this test never ran.

Comment (ammirate) [reply tests/unit/test_reference_builder.py]: fair enough

**Pull request merged**